include/envoy/network/connection.h

```diff
@@ -162,6 +173,17 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * Get the value set with setReadBufferLimit.
    */
   virtual uint32_t readBufferLimit() const PURE;
+
+  /**
+   * Sets the high and low watermarks which trigger onAboveWriteBufferHighWatermark
+   * and onBelowWriteBufferHighWatermark callbacks.
+   * @param low_watermark if the connection was above the high watermark and the
+   * connection buffer is drained below this many bytes, onBelowWriteBufferHighWatermark will be
+   * called.
+   * @param high_watermark if the connection has more bytes than this buffered,
+   * onAboveWriteBufferHighWatermark will be called.
+   */
+  virtual void setWriteBufferWatermarks(size_t low_watermark, size_t high_watermark) PURE;
```

Comment:
nit: prefer fixed sizes in Envoy code (probably uint32_t here but uint64_t is fine if you want).